### ISSUE TITLE
wall_limit not working as expected

### ISSUE THREAD

**Issue by dschwoerer** [boutproject/BOUT-dev#1557]
I have noticed in one run that wall_limit did not trigger on all threads.
Some files have:
```
6.000e+01      25860       5.72e+03    56.9    2.0    1.7    0.0   39.4
Only 1.022924e+04 seconds (1.79 steps) left. Quitting
Monitor signalled to quit
Error encountered in solver run
```
But some have just
```
6.000e+01      25860       5.72e+03    57.1    2.0    1.6    0.0   39.3
```
I didn't notice at first, because everything seemed fine - but things got strange when I restarted the simulation:
```
Run started at  : Wed Feb  6 10:07:20 2019

        Option :restart = true (Command line)
        Option :append = true (Command line)
        Option :dump_on_restart = false  (default)
slurmstepd: error: *** JOB 3575425 ON r145c05s02 CANCELLED AT 2019-02-07T10:07:32 DUE TO TIME LIMIT ***
slurmstepd: error: *** STEP 3575425.0 ON r145c05s02 CANCELLED AT 2019-02-07T10:07:32 DUE TO TIME LIMIT ***
```
i.e. the simulation was doing nothing for 24 hours.

I am wondering whether the monitor triggered an hard-quit, and thus some processes got killed before they finished writing - and thus the dump files are in an inconsistent state? Preventing the simulation from running?

This is an older version of next, but I am not sure how to reproduce ...

**Comment by dschwoerer** [comment]
Fixed by #1558 in next.
Should it also be fixed in master?

**Comment by dschwoerer** [comment]
Fixed by #1610 in master
